src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+public class Intake extends SubsystemBase {
+    private static Intake INSTANCE;
+    private final WPI_TalonSRX motor = new WPI_TalonSRX(Ports.Intake.MOTOR);
+    private final Solenoid solenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Intake.SOLENOID);
+
+    private Intake() {
+        motor.setInverted(Ports.Intake.IS_INVERTED);
+    }
+
+
+    /**
+     * Lazy instantiation.
+     *
+     * @return the subsystem instance.
+     */
+    public static Intake getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Intake();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Sets the power of the intake [%].
+     *
+     * @param power desired power in percentage.
+     */
+    public void setPower(double power) {
+        motor.set(ControlMode.PercentOutput, power);
+    }
+
+    /**
+     * Open intake solenoid.
+     */
+    public void openSolenoid() {
+        solenoid.set(Constants.Intake.IS_SOLENOID_INVERTED);
+    }
+
+    /**
+     * Close intake solenoid.
+     */
+    public void closeSolenoid() {
+        solenoid.set(!Constants.Intake.IS_SOLENOID_INVERTED);
+    }

```

Comment:
1) The solenoid is a hardware detail and should not be exposed outside. `open` and `close` are simpler and better IMO (other names might fit too).
2) Following my comment above about the constants, use `true` and `false` here.